src/sensor/ping360.cpp

```diff
@@ -99,19 +100,17 @@ Ping360::Ping360()
     });
 
     connect(this, &Ping360::firmwareVersionMinorChanged, this, [this] {
-        qCDebug(PING_PROTOCOL_PING360) << "Firmware version:"
-                                       << _commonVariables.deviceInformation.firmware_version_major
-                                       << _commonVariables.deviceInformation.firmware_version_minor
-                                       << _commonVariables.deviceInformation.firmware_version_patch;
+        QVersionNumber version(_commonVariables.deviceInformation.firmware_version_major,
+            _commonVariables.deviceInformation.firmware_version_minor,
+            _commonVariables.deviceInformation.firmware_version_patch);
+        qCDebug(PING_PROTOCOL_PING360) << "Firmware version:" << version;
 
         // Wait for firmware information to be available before looking for new versions
         static bool once = false;
         if (!once && _commonVariables.deviceInformation.initialized) {
             once = true;
 
-            if (_commonVariables.deviceInformation.firmware_version_major == 3
-                && _commonVariables.deviceInformation.firmware_version_minor == 3
-                && _commonVariables.deviceInformation.firmware_version_patch == 1) {
+            if (version > QVersionNumber(3, 3, 0)) {
```

Comment:
Wait, we changed here from 3.3.1 to 3.3.0!

Reply:
hmm, it was `==`, not `>=`. Is this okay?

Reply:
That's the intended behavior, the async feature is only available on newer firmware, that are newer than 3.3.0, it started on 3.3.1

Reply:
okay